plugins/out_kinesis_firehose/firehose.c

```diff
@@ -119,6 +121,15 @@ static int cb_firehose_init(struct flb_output_instance *ins,
         ctx->sts_endpoint = (char *) tmp;
     }
 
+    tmp = flb_output_get_property("compression", ins);
+    if (tmp) {
+        ret = flb_aws_compression_get_type(tmp);
+        if (ret == -1) {
+            flb_plg_error(ctx->ins, "unknown compression: %s", tmp);
+            return -1;
```

Comment:
should we do `goto error` here?

Reply:
Yep. I copied the code from S3 which works but maybe doesn't properly clean up the memory. I'll change to `goto`.

Reply:
Resolved.